app/static/js/slcManagement.js

```diff
@@ -140,3 +187,113 @@ function unapproveProposal(el){
     })
 }
+
+/************** Imported Courses Modal Functions **************/
+function showAlterModalWithCourse(courseID) {
+  getImportedCourseInfo(courseID, function() {
+    $('#alterModal #alterCourseId').val(courseID);
+
+    $('#alterModal').modal('show');
+  });
+
+  $('#saveButton').on('click', function(event) {
+    event.preventDefault();
+    var instructorData = getCourseInstructors();
+    // Assuming you need to clear specific instructor inputs for some reason before proceeding.
+    $('#alterModal form input[name="instructor[]"]').remove();
+    updateInstructorList(instructorData);
+
+    var dynamicRoute = `/manageServiceLearning/imported/${courseID}`;
+    var $form = $('#alterModal form');
+    $form.attr('action', dynamicRoute);
+    $form.submit();
+});
+}
+
+
+function getImportedCourseInfo(courseID, callback) { // This function populates the fields in the modal of a chosen course with preexisting data
+  $.ajax({
+    url: `/manageServiceLearning/imported/${courseID}`,
+    type: "GET",
+    success: function(courseDict) {
+      if (Object.keys(courseDict).length !== 0){
+        // update the imported course modal
+        $('#courseName').val(courseDict['courseName']);
+        $('#courseAbbreviation').val(courseDict['courseAbbreviation']);
+
+        let modalTitle = courseDict['courseName'] ? courseDict['courseName'] : courseDict['courseAbbreviation'] 
+        
+        // Find the element with class "modal-header" and "fw-bold" using jQuery
+        $("#importedModalTitle").text("Edit " + modalTitle);
+
+        $('#hoursEarned').val(courseDict['hoursEarned']);
+        if (courseDict['instructors'] && courseDict['instructors'].length > 0) {
+          updateInstructorsTable(courseDict['instructors']);
+        }
+      }
+      
+      if (callback) callback();
+    } 
+  });
+}
+
+// Instructor manipulation functions
+
+function updateInstructorList(instructorData) { // This function fetches instructor usernames and attached the list of usernames to the form submission
+  $('#alterModal form input[name="instructor[]"]').remove();
+
+  // Append new hidden inputs for each instructor
+  for (let i=0; i < instructorData.length; i++) {
+      $('<input>').attr({
+          type: 'hidden',
+          name: 'instructor[]',
+          value: instructorData[i]
+      }).appendTo('#alterModal form');
+  };
+}
+
+function updateInstructorsTable(instructors) { // This function creates row(s) for preexisting instructor(s) in the modal
+  // Clear existing table contents except the template row
+  $("#instructorTableBody").find("tr:not(:first)").remove();
+  $("#instructorTableNames").empty();
+  // Add each instructor to the table
+  instructors.forEach(function(instructor) {
+    var newRow = createInstructorRow(instructor);
+    $("#instructorTableBody").append(newRow);
+
+    $('<input>').attr({
+      type: 'hidden',
+      name: 'instructor[]',
+      value: instructor.username
+    }).appendTo('#instructorTableNames');
```

Comment:
Why are you maintaining two different lists of the same information? the other list is on line 251. If you need to maintain two lists then you should have a function that does the list stuff and call it where you need it rather than having the same code in two separate spots.

Reply:
Fixed